File: backend/src/services/otherAiModelsService.ts
import { AppDataSource } from "../config/data-source";
import { OtherAiModel } from "../entities/OtherAiModel";

/** The known-models picklist for Other AI Providers (Configuration > AI
 * Assistant > Other AI Providers), sortOrder ascending — read-only reference
 * data, not something the admin curates the way FabrixModel is (these are
 * Anthropic's own official model ids, a small well-known set; the Model
 * field stays free-text-capable in the UI regardless, so a newer id than
 * this seeded list still works). */
export function listOtherAiModels(): Promise<OtherAiModel[]> {
  return AppDataSource.getRepository(OtherAiModel).find({ order: { sortOrder: "ASC" } });
}

File: backend-py/app/services/other_ai_models_service.py
"""Port of `backend/src/services/otherAiModelsService.ts`."""

from __future__ import annotations

from sqlalchemy import select
from sqlalchemy.orm import Session

from app.models.other_ai_model import OtherAiModel


def list_other_ai_models(db: Session) -> list[OtherAiModel]:
    """The known-models picklist for Other AI Providers, sortOrder
    ascending — read-only reference data, not something the admin curates
    the way FabrixModel is."""
    return list(db.execute(select(OtherAiModel).order_by(OtherAiModel.sortOrder.asc())).scalars().all())
